image-rs/src/unpack.rs

```diff
@@ -60,6 +130,10 @@ pub async fn unpack<R: AsyncRead + Unpin>(input: R, destination: &Path) -> Resul
             .try_into()
             .context("GID is too large!")?;
 
+        if !convert_whiteout(&file.path()?, file.header(), destination)? {
```

Comment:
I noticed that there are some duplicated logic between `convert_whiteout` and `unpack` after line 139. Could we do some refactoring upon this commit to avoid this?

Reply:
Yes, I will refactor it

Reply:
@Xynnn007 I pushed some changes. Is that what you have in mind?